resources/helm/dask-gateway/values.schema.yaml

```diff
@@ -108,6 +108,25 @@ properties:
           See [the Kubernetes
           documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/)
           for more details about annotations.
+      env: &env-spec
+        type: array
+        items:
+          type: object
+          additionalProperties: false
+          properties:
+            name:
+              type: string
+              description: |
+                The name of the environment variable to pass to the pod
+            value:
+              type: string
+              description: |
+                The value of the environment variable to pass to the pod
+          required:
+            - name
+            - value
+        description: |
+          Additional environment variables to apply to the resource
```

Comment:
This should allow for configuration as declared in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#envvar-v1-core, including `name` (string), `value` (string), and `valueFrom` (object).

I'd like to see that:
- the description mention that users should use the k8s official configuration syntax
- properties entries for name/value/valueFrom declaring them string/string/object, you can skip the descriptions here to defer to the k8s official docs instead
- for the valueFrom object, just declare `additionalProperties: true` and don't declare its properties
- remove the `&env-spec` anchor definition if its not used elsewhere
- Only `name` is required as otherwise you can't choose from either `value` or `valueFrom`.